File: bank-service/src/main/java/rs/edu/raf/dto/CreditRequestCreateDto.java
package rs.edu.raf.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.PositiveOrZero;

import java.math.BigDecimal;

public record CreditRequestCreateDto (
    @NotNull Long creditTypeId,
    @NotNull @Positive BigDecimal amount,
    @NotNull Long currencyId,
    @NotNull String loanPurpose,
    @NotNull @PositiveOrZero BigDecimal salary,
    @NotNull String phoneNumber,
    boolean permanentEmployee,
    int currentEmploymentPeriod,
    int loanTerm,
    @NotNull Long bankAccountNumber

) { }

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/dto/CreditRequestCreateDTO.kt
package org.example.bankservicekotlin.dto

import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Positive
import jakarta.validation.constraints.PositiveOrZero
import java.math.BigDecimal

data class CreditRequestCreateDTO(
    @NotNull val creditTypeId: Long,
    @NotNull @Positive val amount: BigDecimal,
    @NotNull val loanPurpose: String,
    @NotNull @PositiveOrZero val salary: BigDecimal,
    @NotNull val phoneNumber: String,
    val permanentEmployee: Boolean,
    val currentEmploymentPeriod: Int,
    val loanTerm: Long,
    @NotNull val bankAccountNumber: Long
)
